src/needle.ts

```diff
@@ -107,6 +107,10 @@ export async function makeRequest<T = void>(
       emitter.apiRequestLog(`Requested url --> ${url} | error --> ${err}`);
     }
 
+    const errorMessage = new Object(response?.body)['error'];
```

Comment:
`nit`: better to keep it simple and explicitly override rules IMO
```suggestion
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
    const errorMessage = response?.body?.error;
```